test/units/TEST-50-DISSECT.dissect.sh

```diff
@@ -81,8 +81,40 @@ systemd-run --wait -p RootImage="$MINIMAL_IMAGE.raw" mountpoint /run/systemd/jou
 
 # Test that the notify socket is bind mounted to /run/host/notify in sandboxed environments and
 # $NOTIFY_SOCKET is set correctly.
-systemd-run --wait -p RootImage="$MINIMAL_IMAGE.raw" -p NotifyAccess=all --service-type=notify --pipe sh -c 'printf MAINPID=$$$$\\nREADY=1 | ncat --unixsock --udp $NOTIFY_SOCKET --source /run/notify && test -S /run/host/notify'
-systemd-run --wait -p RootImage="$MINIMAL_IMAGE.raw" -p NotifyAccess=all --service-type=notify --pipe sh -c 'printf MAINPID=$$$$\\nREADY=1 | ncat --unixsock --udp $NOTIFY_SOCKET --source /run/notify && env' | grep NOTIFY_SOCKET=/run/host/notify
+systemd-run \
+    --wait \
+    -p RootImage="$MINIMAL_IMAGE.raw" \
+    -p NotifyAccess=all \
+    --service-type=notify \
+    --pipe \
+    bash -xec \
+    '
+        printf MAINPID=$$$$\\nREADY=1 | ncat --unixsock --udp $NOTIFY_SOCKET --source /run/notify
+        [[ "$$NOTIFY_SOCKET" == "/run/host/notify" ]]
+        [[ "$$(env)" =~ "NOTIFY_SOCKET=/run/host/notify" ]]
+        test -S /run/host/notify
+    '
+if [[ "$(findmnt -n -o FSTYPE /)" == btrfs ]]; then
+    [[ -d /test-dissect-btrfs-snapshot ]] && btrfs subvolume delete /test-dissect-btrfs-snapshot
+    btrfs subvolume snapshot / /test-dissect-btrfs-snapshot
+
+    # Same test with systemd-notify and RootDirectory=
+    systemd-run \
+        --wait \
+        -p RootDirectory=/test-dissect-btrfs-snapshot \
```

Comment:
You can move this further down and use `RootDirectory=/tmp/img` instead - that way we avoid btrfs shenanigans. It's a bit crashy lately...

Reply:
/tmp/img is from the minimal image and it does not contain systemd-notify, right?

Reply:
ah you wanted to use that - never mind then